Source/WebCore/dom/mac/ImageControlsMac.cpp

```diff
@@ -76,17 +76,35 @@ bool hasControls(const HTMLElement& element)
     return shadowRoot->hasElementWithId(*imageControlsElementIdentifier().impl());
 }
 
+static RefPtr<HTMLElement> imageControlsHost(const Node& node)
+{
+    RefPtr host = dynamicDowncast<HTMLElement>(node.shadowHost());
+    if (!host)
+        return nullptr;
+
+    return hasImageControls(*host) ? host : nullptr;
```

Comment:
Don't we want to check the type of host (UA shadow root as opposed to any shadow root)?